src/internal/serde/interface.go

```diff
@@ -138,40 +138,3 @@ type Encoder interface {
 	// TODO(msteffen) same comment re: proto.Message as for EncodeProto()
 	EncodeProtoTransform(proto.Message, func(map[string]interface{}) error) error
 }
-
-// Decoder is the interface for decoding a particular serialization format.
-// Currently, the only implementation of Decoder is in serde/yaml/decoder.go
-// (the point of having a common interface is to make it possible to switch
-// between decoding different serialization formats at runtime, but we have no
-// need to switch between a YAML decoder and a JSON decoder, as our YAML decoder
-// can decode JSON).
-type Decoder interface {
```

Comment:
This interface unified JSON and YAML decoding for PipelineManifestReader, but since we're switching to YAML parsing only, there's only one implementation of the interface and it's no longer really necessary.